R/one_call_pipe_linter.R

```diff
@@ -0,0 +1,78 @@
+#' Block single-call magrittr pipes
+#'
+#' Prefer using a plain call instead of a pipe with only one call,
+#'   i.e. `1:10 %>% sum()` should instead be `sum(1:10)`. Note that
+#'   calls in the first `%>%` argument count. `rowSums(x) %>% max()` is OK
+#'   because there are two total calls (`rowSums()` and `max()`).
+#'
+#' Note also that un-"called" steps are *not* counted, since they should
+#'   be calls (see [pipe_call_linter()]).
+#'
+#' @examples
+#' # will produce lints
+#' lint(
+#'   text = "(1:10) %>% sum()",
+#'   linters = one_call_pipe_linter()
+#' )
+#'
+#' lint(
+#'   text = "DT %>% .[grp == 'a', sum(v)]",
+#'   linters = one_call_pipe_linter()
+#' )
+#'
+#' # okay
+#' lint(
+#'   text = "rowSums(x) %>% mean()",
+#'   linters = one_call_pipe_linter()
+#' )
+#'
+#' lint(
+#'   text = "DT[src == 'a', .N, by = grp] %>% .[N > 10]",
```

Comment:
Seems like a bad example because the pipe is unnecessary alltogether.

Reply:
more realistically the second step would come on the next line, but such usage is not unheard of, e.g.

https://github.com/TheEconomist/big-mac-data/blob/54c2663f5237c9bdbd34c99cf1ec4f87062b6469/data-generator.R#L18

Reply:
I would put `][` on a new line, but fine.

Reply:
I've found [] chaining is not v friendly with tidy syntax (or styler at least), and mixing data.table steps with tidyverse (e.g. dplyr, tidyr, or ggplot2) steps can be more natural to use this form of chaining.